ts/main.ts

```diff
@@ -183,13 +192,15 @@ async function fetchExerciseSearchData(term: string): Promise<void> {
     }
     if (exerciseObjArr.length > 0) {
       exerciseObjArr.forEach((element) => {
-        $cardList.appendChild(renderExercises(element));
+        $exercisesCardList.appendChild(renderExercises(element));
       });
       $noResults?.classList.add('hidden');
     } else {
       $noResults?.classList.remove('hidden');
     }
-    $exercisesNodeList = document.querySelectorAll('.card');
+    $exercisesNodeList = document.querySelectorAll(
+      '#exercises-card-list > .card',
+    );
     if (!$exercisesNodeList) throw new Error('no exercise nodelist found');
   } catch (error) {
```

Comment:
As part of the above comment, if you're not using using the NodeList here, don't query for it until you need it. I would put your query in your `handleFavoriteClick` function instead